### PR TITLE
Slim up the contract

### PR DESCRIPTION
**Pull request by kalepail**
Don't store signatures in an unbound vector. Instead let's use Mercury to track some new events.

### CODE REVIEW

File: contracts/contract-webauthn-secp256r1/src/lib.rs [hunk 1]
```diff
@@ -219,20 +188,20 @@ impl CustomAccountInterface for Contract {
...
                             || c.fn_name == Symbol::new(&env, "upgrade"))
                     {
                         if env
                             .storage()
                             .instance()
-                            .get::<Symbol, Bytes>(&SUDO_SIGNER)
+                            .get::<Symbol, Bytes>(&SUPER)
                             .ok_or(Error::NotFound)?
                             != signature.id
                         {
```
Comment (dmkozh) [inline contracts/contract-webauthn-secp256r1/src/lib.rs:207]: `_ => return Err(Error::InvalidContext),` - that's disallows deploying new contracts by this account, which seems like a weird/arbitrary restriction to me.

File: contracts/contract-webauthn-secp256r1/src/lib.rs [hunk 2]
```diff
@@ -219,20 +188,20 @@ impl CustomAccountInterface for Contract {
         signature: Signature,
         auth_contexts: Vec<Context>,
     ) -> Result<(), Error> {
-        // Only the sudo signer can `add_sig`, `rm_sig`, `resudo` and `upgrade`
+        // Only the super signer can `add_sig`, `rm_sig`, `re_super` and `upgrade`
         for context in auth_contexts.iter() {
             match context {
                 Context::Contract(c) => {
                     if c.contract == env.current_contract_address()
                         && (c.fn_name == Symbol::new(&env, "add_sig")
```
Comment (dmkozh) [inline contracts/contract-webauthn-secp256r1/src/lib.rs:196]: nit: Prefer `symbol_short!` for clarity and compile-time check that your symbols are indeed short.

File: contracts/contract-webauthn-secp256r1/src/lib.rs (hunk 1 above)
Comment (dmkozh) [inline contracts/contract-webauthn-secp256r1/src/lib.rs:207]: `.extend_ttl(&signature.id, max_ttl, max_ttl);` - I'm not sure if always setting the threshold to max ttl is a good idea; it makes frequent transactions to unnecessarily spend a bit more fees on extension without much benefit (say, your first transaction extends the contract to 120 days, then the second transaction extends it for 10 more minutes. is it really worth it?). I'd suggest using max_ttl - a few days in order to reduce the bump frequency. That's worth changing across all the TTL extensions in these contracts.

Comment (kalepail) [reply contracts/contract-webauthn-secp256r1/src/lib.rs:212]: Are there any concerns such that only the super signer should be allowed to deploy new contracts? Probably not. So this should just be `_ => {}` to enable it correct? Essentially pass over that instance.

Comment (kalepail) [reply contracts/contract-webauthn-secp256r1/src/lib.rs:231]: Made some adjustments, is that what you had in mind? Max extend but with a 7 day max threshold.

Comment (dmkozh) [reply contracts/contract-webauthn-secp256r1/src/lib.rs:231]: Yeah, that's good

Comment (dmkozh) [reply contracts/contract-webauthn-secp256r1/src/lib.rs:231]: Yes, just _ => {} is good, my point is that we shouldn't really return an error for authorizing the host functions.

**Pull request merged**